CMakeLists.txt

```diff
@@ -26,8 +26,90 @@ include_directories(include/)
 
 set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/deps/findicu)
 find_package(Threads REQUIRED)
-find_package(ICU COMPONENTS data i18n uc REQUIRED)
-include_directories(SYSTEM ${ICU_INCLUDE_DIRS})
+find_package(ICU 56.1 COMPONENTS data i18n uc)
+
+if (NOT ICU_FOUND OR ICU_VERSION VERSION_LESS 56.1)
```

Comment:
Yeah, let me do that. I should probably start encapsulating parts of the main `CMakeLists.txt` since it's getting rather long, too... so let me do that and then I'll ping you again for review.